plugin/clouddns/setup.go

```diff
@@ -98,7 +98,7 @@ func setup(c *caddy.Controller) error {
 			h.Next = next
 			return h
 		})
-		c.OnShutdown(func() error { ctx.Done(); return nil })
```

Comment:
`Done()` should to the exact same thing as `cancel()` so this PR should be a noop.

Maybe `ctx.Done()` is called wrong, and reading the doc I see we should call `close(ctx.Done())` (although I'm not super clear after reading it for 5min). Look like cancel is a better and more readable way to do this.